### PR TITLE
1.14.6 rebase

### PR DESCRIPTION
**Pull request by deads2k**
About 6 hours

Updates to 1.14.6 and eliminates patches that are only needed by other repos.

Picklist: https://docs.google.com/spreadsheets/d/12txg87Jq-BRfPsgMYWgdy_eH94lVOwGtseIe4OWPmZ0/edit#gid=1413655187

### CODE REVIEW

**Comment by deads2k** [comment]
/hold

to ensure this isn't merged until we re-check the carry list

**Comment by derekwaynecarr** [comment]
agree we should take the latest upstream prior to 4.2.0.

**Comment by mfojtik** [comment]
/lgtm

Please open a follow up BZ for the file locking issue

/hold

Holding so @soltysh and @sttts can take a stab

File: vendor/k8s.io/kubernetes/pkg/kubectl/cmd/config/config_test.go [hunk 1]
```diff
@@ -87,6 +87,7 @@ func TestCurrentContext(t *testing.T) {
 }
 
 func TestSetCurrentContext(t *testing.T) {
+	t.Skip()
```
Comment (sttts) [inline vendor/k8s.io/kubernetes/pkg/kubectl/cmd/config/config_test.go]: document in the commit message why

File: glide.yaml [hunk 2]
```diff
@@ -71,7 +71,7 @@ import:
   version: origin-4.2-kubernetes-1.14.0
```
Comment (sttts) [inline glide.yaml]: while probably not super important. Would switch all of them.

**Comment by sttts** [comment]
Looks good overall. Some nits.

**Comment by deads2k** [comment]
/hold

holding again because there's a new upstream commit

File: vendor/k8s.io/kubernetes/pkg/kubectl/cmd/config/config_test.go (hunk 1 above)
Comment (deads2k) [reply vendor/k8s.io/kubernetes/pkg/kubectl/cmd/config/config_test.go]: > document in the commit message why

removed that commit.  Found the lock file culprit

**Comment by deads2k** [comment]
/hold cancel

updated

**Pull request merged**